packages/lib/src/toolbar/controls/CellWidthInput.tsx

```diff
@@ -0,0 +1,76 @@
+import { useDebouncedCallback } from '@react-hookz/web';
+import { useState } from 'react';
+import { FiRotateCw } from 'react-icons/fi';
+
+import Btn from './Btn';
+import styles from './CellWidthInput.module.css';
+
+const ID = 'h5w-cell-width';
+const MIN = 50;
+
+interface Props {
+  value: number | undefined;
+  defaultValue: number;
+  onChange: (val: number | undefined) => void;
+}
+
+function CellWidthInput(props: Props) {
+  const { value, defaultValue, onChange } = props;
+  const hasValue = value !== undefined;
+
+  const [inputValue, setInputValue] = useState(
+    (value ?? defaultValue).toString()
+  );
+
+  const onChangeDebounced = useDebouncedCallback(onChange, [onChange], 150);
+
+  return (
+    <div className={styles.root}>
+      <label id={`${ID}-label`} className={styles.label} htmlFor={ID}>
+        Cell width
+      </label>
+
+      <input
+        id={ID}
+        className={styles.input}
+        type="number"
+        value={inputValue}
+        step={10}
+        min={MIN}
+        placeholder={defaultValue.toString()}
+        aria-labelledby={`${ID}-label`} // https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/566
+        onInput={(evt) => {
+          const { value } = evt.currentTarget;
+          setInputValue(value);
+
+          const numValue = Number.parseInt(value);
+          const safeNumValue = Number.isNaN(numValue)
+            ? defaultValue
+            : numValue < MIN
+            ? MIN
+            : numValue;
+
+          onChangeDebounced(
+            safeNumValue === defaultValue ? undefined : safeNumValue
+          );
+        }}
+        onBlur={() => setInputValue((value ?? defaultValue).toString())}
```

Comment:
On blur, we clean things up by overriding the input value with the custom value. This is useful if the user enters an invalid value or clears the input for instance, in which case the value gets reset to `undefined` in `onInput`. If the user enters a value lower than `MIN`, `onBlur` sets it to `MIN`.